spec/datadog/core/remote/worker_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/core/remote/worker'
+
+RSpec.describe Datadog::Core::Remote::Worker do
+  let(:task) { proc { 1 + 1 } }
+  subject(:worker) { described_class.new(interval: 1, &task) }
+
+  describe '#initialize' do
+    it 'raises ArgumentError when no block is provided' do
+      expect do
+        described_class.new(interval: 1)
+      end.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#start' do
+    after { worker.stop }
+
+    it 'mark worker as started' do
+      expect(worker).not_to be_started
+      worker.start
+      expect(worker).to be_started
+    end
+
+    it 'acquire and release lock' do
+      expect(worker).to receive(:acquire_lock).at_least(:once)
+      expect(worker).to receive(:release_lock).at_least(:once)
```

Comment:
`at_least(:once)` because calling `worker.stop`, which happen on the after block, will also call `acquire_lock && release_lock`